library/src/G4CMPSolidUtils.cc

```diff
@@ -0,0 +1,392 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+/// \file library/src/G4PhononReflection.cc
+/// \brief This class contains custom utilities for G4VSolids that are not
+/// directly included in the G4VSolid class. Some example of utilites are:
+/// 1) Adjusting a position to the nearest surface point without the surface normal.
+/// 2) Adjusting a particle to the nearest edge along the surface.
+/// 3) Reflecting a tangent vector against a "hard" edge.
+///
+/// The user will need to create this class directly with a G4VSolid (and transform) or
+/// a touchable. If the transform or touchable is not provided, an indentity transform is
+/// used.
+///
+/// Note: All coordinates/vectors passed into functions must be in the global coordinate system.
+//
+// $Id$
+//
+// 20250424  G4CMP-465 -- Create G4CMPSolidUtils class.
+
+#include "G4CMPSolidUtils.hh"
+#include "G4AffineTransform.hh"
+#include "G4PhysicalConstants.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4VSolid.hh"
+#include "G4VTouchable.hh"
+#include "G4UnitsTable.hh"
+
+
+// Direct constructors
+
+G4CMPSolidUtils::G4CMPSolidUtils(const G4VSolid* solid=0,
+                                 G4int verbose=0,
+                                 G4String vLabel="G4CMPSolidUtils")
+  : theSolid(solid),
+    theTransform(G4AffineTransform(G4RotationMatrix::IDENTITY, G4ThreeVector(0,0,0))),
```

Comment:
You don't need to do this explicitly.  `G4AffineTransform()` by itself returns the identity.